data/shared/citizen/scripting/lua/scheduler.lua

```diff
@@ -280,23 +275,27 @@ Citizen.SetEventRoutine(function(eventName, eventPayload, eventSource)
 
 	-- try finding an event handler for the event
 	local eventHandlerEntry = eventHandlers[eventName]
-	
-	-- deserialize the event structure (so that we end up adding references to delete later on)
-	local data = msgpack.unpack(eventPayload)
 
 	if eventHandlerEntry and eventHandlerEntry.handlers then
 		-- if this is a net event and we don't allow this event to be triggered from the network, return
 		if eventSource:sub(1, 3) == 'net' then
-			if not eventHandlerEntry.safeForNet and not alwaysSafeEvents[eventName] then
+			if not eventHandlerEntry.safeForNet then
 				Citizen.Trace('event ' .. eventName .. " was not safe for net\n")
 
+				_G.source = lastSource
 				return
 			end
 
 			deserializingNetEvent = { source = eventSource }
 			_G.source = tonumber(eventSource:sub(5))
+		elseif IsDuplicityVersion() and eventSource:sub(1, 12) == 'internal-net' then
```

Comment:
Do we not cache IsDuplicityVersion's result elsewhere already? Since this won't change at runtime, calling _this_ frequently is a bit redundant.

Reply:
No it is not cached anywhere however, as you said, it won't change at runtime so caching it will improve performance not just here but in other places like exports, RPC invocation and state bags. I'll make this into its own commit, separate from the other changes requested in this file but including the same requested changes to the JS ScRT, grouping the same type of changes together.